content_handler/framebuffer_skia.cc

```diff
@@ -0,0 +1,65 @@
+// Copyright 2016 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/content_handler/framebuffer_skia.h"
+
+#include <magenta/syscalls.h>
+
+#include <utility>
+
+#include "lib/ftl/logging.h"
+
+namespace flutter_content_handler {
+
+FramebufferSkia::FramebufferSkia() {}
+
+FramebufferSkia::~FramebufferSkia() {}
+
+void FramebufferSkia::Bind(mojo::InterfaceHandle<mojo::Framebuffer> framebuffer,
+                           mojo::FramebufferInfoPtr info) {
+  if (!framebuffer) {
+    FTL_LOG(ERROR) << "Failed to bind framebuffer";
+    surface_ = nullptr;
+    return;
+  }
+
+  framebuffer_.Bind(std::move(framebuffer));
+  info_ = std::move(info);
+
+  uintptr_t buffer = 0;
+  size_t row_bytes = info_->row_bytes;
+  size_t size = row_bytes * info_->size->height;
+
+  mx_status_t status =
+      mx_process_vm_map(0, info_->vmo.get().value(), 0, size, &buffer,
+                        MX_VM_FLAG_PERM_READ | MX_VM_FLAG_PERM_WRITE);
+
+  if (status < 0) {
+    FTL_LOG(ERROR) << "Cannot map framebuffer (status=" << status << ")";
+    return;
+  }
+
+  SkColorType sk_color_type;
+  switch (info_->format) {
+    case mojo::FramebufferFormat::RGB_565:
+      sk_color_type = kRGB_565_SkColorType;
+      break;
+    case mojo::FramebufferFormat::ARGB_8888:
+      sk_color_type = kRGBA_8888_SkColorType;
+      break;
+    default:
+      FTL_LOG(WARNING) << "Unknown color type " << info_->format;
+      sk_color_type = kRGB_565_SkColorType;
+      break;
+  }
+
+  SkImageInfo image_info =
+      SkImageInfo::Make(info_->size->width, info_->size->height, sk_color_type,
+                        kPremul_SkAlphaType);
```

Comment:
In case, the `kRGB_565_SkColorType` format, `kPremul_SkAlphaType` will be incorrect.  I suggest an `SkAlphaTypeIsOpaque` check like the one performed in https://github.com/flutter/engine/blob/b2058f8b3d19bcf1745397b568f4787daff32d04/flow/texture_image_gles2.cc#L196.